src/test/cc/wfanet/panelmatch/common/crypto/aes_test.cc

```diff
@@ -16,34 +16,183 @@
   std::unique_ptr<Aes> aes = GetAesSivCmac512();
-  auto result = aes->Decrypt("input", SecretDataFromStringView("key"));
-  EXPECT_THAT(result.status(), StatusIs(absl::StatusCode::kUnimplemented, ""));
+  SecretData key = GetTestKey();
+  std::string plaintext = "Some data to encrypt.";
+  auto cyphertext = aes->Encrypt(plaintext, key);
+  key = SecretDataFromStringView(absl::HexStringToBytes(""));
+  auto result = aes->Decrypt(*cyphertext, key);
+  EXPECT_THAT(result.status(),
+              StatusIs(absl::StatusCode::kInvalidArgument, ""));
 }
 
-// TODO(efreck): Implement the following test cases
-//  - Encrypt/Decrypt
-//  - Compares the output of AesSiv to AesSivBoringSsl
-//  - Wrong key size
-//  - Null input
-//  - Null key
-//  - Different key with same string are different things
-//  - Same key with different strings are different things
+// Tests that different keys with the same string return different values
+TEST(AesTest, differentKeySameStringEncrypt) {
+  std::unique_ptr<Aes> aes = GetAesSivCmac512();
+  SecretData key_1 = SecretDataFromStringView(absl::HexStringToBytes(
+      "990102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"
+      "99112233445566778899aabbccddeefff0f1f2f3f4f5f6f7f8f9fafbfcfdfeff"));
+  SecretData key_2 = GetTestKey();
+  std::string_view plaintext = "Some data to encrypt.";
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_1,
+                       aes->Encrypt(plaintext, key_1));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_2,
+                       aes->Encrypt(plaintext, key_2));
+  EXPECT_NE(cyphertext_1, cyphertext_2);
+}
+
+// Tests that decrypting with a different key than encryption gives an error
+TEST(AesTest, differentKeyDecrypt) {
+  std::unique_ptr<Aes> aes = GetAesSivCmac512();
+  SecretData key_1 = SecretDataFromStringView(absl::HexStringToBytes(
+      "990102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"
+      "99112233445566778899aabbccddeefff0f1f2f3f4f5f6f7f8f9fafbfcfdfeff"));
+  SecretData key_2 = GetTestKey();
+  std::string_view plaintext = "Some data to encrypt.";
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext, aes->Encrypt(plaintext, key_1));
+  auto decrypted = aes->Decrypt(cyphertext, key_2);
+  EXPECT_THAT(decrypted.status(),
+              StatusIs(absl::StatusCode::kInvalidArgument, ""));
+}
+
+// Tests that the same key with different strings return different values
+TEST(AesTest, sameKeyDifferentStringEncrypt) {
+  std::unique_ptr<Aes> aes = GetAesSivCmac512();
+  SecretData key = GetTestKey();
+  std::string_view plaintext_1 = "Some data to encrypt.";
+  std::string_view plaintext_2 = "Additional data to encrypt.";
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_1,
+                       aes->Encrypt(plaintext_1, key));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_2,
+                       aes->Encrypt(plaintext_2, key));
+  EXPECT_NE(cyphertext_1, cyphertext_2);
+}
+
+// Tests that the same key with different strings return different values
```

Comment:
The comment for this unit test is the same as for the previous one.  What is this supposed to be testing?